web/src/server/api/routers/traces.ts

```diff
@@ -658,54 +663,72 @@ export const traceRouter = createTRPCRouter({
         });
       }
 
-      await ctx.prisma.$transaction([
-        ctx.prisma.trace.deleteMany({
-          where: {
-            id: {
-              in: input.traceIds,
+      if (!traceDeleteQueue) {
+        logger.warn(
+          `TraceDeleteQueue not initialized. Try synchronous deletion for ${input.traceIds.length} traces.`,
+        );
```

Comment:
Moved all of the existing logic into this if block. That keeps it compatible for V2 and allows us a fallback if Redis instantiation doesn't work. Can probably be dropped in the future.